common/src/main/java/com/datastrato/gravitino/dto/requests/TableCreateRequest.java

```diff
@@ -108,5 +110,13 @@ public void validate() throws IllegalArgumentException {
     if (partitioning != null) {
       Arrays.stream(partitioning).forEach(p -> p.validate(columns));
     }
+
+    List<ColumnDTO> autoIncrementCols =
+        Arrays.stream(columns)
+            .peek(ColumnDTO::validate)
+            .filter(ColumnDTO::autoIncrement)
+            .collect(Collectors.toList());
+    Preconditions.checkArgument(
+        autoIncrementCols.size() <= 1, "Only one column can be auto-incremented.");
```

Comment:
Listing the column names of auto-incremented columns in the error message would be more user-friendly.